spec/functional/resource/windows_certificate2_spec.rb

```diff
@@ -0,0 +1,168 @@
+
+  let(:resource) do
+    run_context = Chef::RunContext.new(Chef::Node.new, {}, Chef::EventDispatch::Dispatcher.new)
+    Chef::Resource::WindowsCertificate.new("ChefFunctionalTest", run_context).tap do |r|
+      r.store_name = store
+    end
+  end
+
+  before do
+    # Bypass validation of the store name so we can use a fake test store.
+    allow_any_instance_of(Chef::Mixin::ParamsValidate)
+      .to receive(:_pv_equal_to)
+      .with({ store_name: store }, :store_name, anything)
+      .and_return(true)
+
+      create_store
+
+  end
+
+  after { delete_store }
+
+  describe "action: verify" do
+    it "fails with no certificates in the store" do
+      # expect(Chef::Log).to receive(:info).with("Certificate not found")
+
+      # resource.source = tests_thumbprint
+      # resource.run_action(:verify)
+
+      # expect(resource).not_to be_updated_by_last_action
+    end
+  end
+
+  describe "action: fetch" do
+    # context "with no certificate in the store" do
+    #   it "throws an error with no certificates in the store" do
+    #     expect(Chef::Log).not_to receive(:info)
+    #     resource.source = others_thumbprint
+    #     resource.output_path = cert_output_path
+    #     expect { resource.run_action :fetch }.to raise_error(ArgumentError)
+    #   end
+    # end
+
+    # context "with a certificate in the store" do
+    #   before do
+    #     resource.source = cer_path
+    #     resource.run_action(:create)
+    #   end
+
+    #   it "succeeds with a valid thumbprint" do
+    #     resource.source = tests_thumbprint
+    #     local_output_path = ::File.join(Chef::Config[:file_cache_path], "test.pem")
+    #     resource.output_path = local_output_path
+    #     resource.run_action(:fetch)
+    #     expect(File.exist?(local_output_path)).to be_truthy
+    #   end
+
+    #   it "fails with an invalid thumbprint", :focus do
+    #     expect(Chef::Log).not_to receive(:info)
+
+    #     resource.source = others_thumbprint
+
+    #     Dir.mktmpdir do |dir|
+    #       path = File.join(dir, "test.pem")
+
+    #       resource.output_path = path
+    #       expect { resource.run_action :fetch }.to raise_error(ArgumentError)
+    #     end
+    #   end
+    # end
+
+    context "with a pfx/pkcs12 object in the store" do
+      before do
+        resource.source = pfx_path
+        resource.pfx_password = password
+        resource.exportable = true
+        resource.run_action(:create)
+      end
+
+      # it "verfies" do
+      #   resource.source = tests_thumbprint
+      #   resource.run_action(:verify)
+      # end
+
+      it "exports a PFX file with a valid thumbprint", :focus do
+        resource.source = tests_thumbprint
+        resource.pfx_password = password
+        resource.output_path = pfx_output_path
+        resource.run_action(:fetch)
+        expect(File.exist?(pfx_output_path)).to be_truthy
+      end
+
+      # it "exports a key file with a valid thumbprint" do
```

Comment:
We shouldn't have anything commented out

Reply:
that file shouldn't even be there. UGH... freaking PR's.... that was a test file. Deleting it